### PR TITLE
[mmn] check compress log for MMN

### PR DESCRIPTION
**Pull request by Hikaru2035**
# Ticket
#435

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: cmd/init.go [hunk 1]
```diff
@@ -313,7 +317,63 @@ func initializeFileLogger() {
...
+	defer fw.Close()
+
+	gw := gzip.NewWriter(fw)
+	if _, err := io.Copy(gw, fr); err != nil {
+        gw.Close()
+        return 
+    }
+    
+    gw.Close()
+    fr.Close()
```
Comment (vietdangquoc) [inline cmd/init.go]: call double

File: cmd/init.go [hunk 2]
```diff
@@ -313,7 +317,63 @@ func initializeFileLogger() {
...
+}
+
+func compressLogFile(filePath string) {
+	// Implementation for compressing log file to gzip
+	fr, err := os.Open(filePath)
+	if err != nil { return }
+	defer fr.Close()
+
+	fw, err := os.Create(filePath + ".gz")
+	if err != nil { return }
```
Comment (vietdangquoc) [inline cmd/init.go]: missing log if error

File: cmd/init.go [hunk 3]
```diff
@@ -313,7 +317,63 @@ func initializeFileLogger() {
...
+	defer fw.Close()
+
+	gw := gzip.NewWriter(fw)
+	if _, err := io.Copy(gw, fr); err != nil {
```
Comment (vietdangquoc) [inline cmd/init.go:400]: Nếu io.Copy thất bại giữa chừng (disk đầy, I/O error...), file .gz được tạo ra nhưng nội dung không đầy đủ và không được xóa. Lần sau compressor chạy lại, file .log gốc vẫn còn nhưng đã có file .gz lỗi tồn tại

File: cmd/init.go [hunk 4]
```diff
@@ -313,7 +317,63 @@ func initializeFileLogger() {
...
+		
+		for _, f := range files {
+			if f.IsDir() { continue }
+
+			filePath := filepath.Join(logDir, f.Name())
+			info, err := f.Info()
+			if err != nil { continue }
+
+			if strings.HasSuffix(f.Name(), ".log") && f.Name() != "node1.log" && f.Name() != "mmn-bootnode.log" {
+				if time.Since(info.ModTime()) > 30 *time.Minute {
```
Comment (vietdangquoc) [inline cmd/init.go]: Nếu node idle > 30 phút, mmn.log bị nén và xóa trong khi lumberjack vẫn giữ file handle → mất log hoàn toàn từ thời điểm đó.

**Comment by Hikaru2035** [comment]
Cherry-pick branch: https://github.com/mezonai/mmn/pull/528
